sys/kern/src/arch/riscv32.rs

```diff
@@ -164,6 +164,10 @@ impl task::ArchState for SavedState {
 // scaler that enables ITM, and because ITM is particularly useful when
 // debugging boot failures, this should be set as early in boot as it can
 // be.
+/// # Safety
+///
+/// In a single-threaded, single-process context (which the kernel is in),
+/// access to global mutables are safe as data races are impossible.
```

Comment:
In that case, why mark this function as unsafe? My understanding is that if a function is marked as unsafe, then the caller is responsible for ensuring that it meets the requirements for calling the function safely. [Reference](https://std-dev-guide.rust-lang.org/documentation/safety-comments.html)

Reply:
Given that the kernel is currently restricted to being single-threaded, I think it's fine to just mark the function as safe.